### PR TITLE
Fix warnings with gcc >5.2 and Qt>=5.7

### PR DESCRIPTION
**Pull request by oberon2007**
- fix pch and deprecated qt declarations
- adjust misleading indenting
- deal with unused param 'truncate' in durationtype.cpp
- simpify some int casts

replaces https://github.com/musescore/MuseScore/pull/3183

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -163,8 +163,8 @@ else (APPLE)
       set(CMAKE_CXX_FLAGS_RELEASE "-std=c++11 -mno-ms-bitfields -O2 -DNDEBUG -DQT_NO_DEBUG")
       set(CMAKE_EXE_LINKER_FLAGS "-Wl,--large-address-aware")
    else (MINGW)
-      set(CMAKE_CXX_FLAGS_DEBUG   "-std=c++11 -fPIC -g")
-      set(CMAKE_CXX_FLAGS_RELEASE "-std=c++11 -fPIC -O2 -DNDEBUG -DQT_NO_DEBUG")
+      set(CMAKE_CXX_FLAGS_DEBUG   "-std=gnu++11 -fno-inline -fPIC -g")
+      set(CMAKE_CXX_FLAGS_RELEASE "-std=gnu++11 -fno-inline -fPIC -O2 -DNDEBUG -DQT_NO_DEBUG")
```
Comment (lasconic) [inline CMakeLists.txt]: Why is that needed? For Debug, ok. But for release, no inline means we will loose some optimization. That's not good :-/

Comment (oberon2007) [inline CMakeLists.txt]: The reason was that I see a lot of these warnings:
```
all.h.gch: not used because `__NO_INLINE__' not defined [-Winvalid-pch]
```
Can this be fixed in **all.h.gch** maybe?

Comment (lasconic) [inline CMakeLists.txt]: Are you 100% sure that you are not using a all.h.gch generated during a debug build to compile a release version? 
Also maybe check the compilation of the precompiled header with make VERBOSE=1 and make sure that the compilation flags are the same for this compilation and for the compilation of any file with this warnings. In particular the -O flag.

Comment (oberon2007) [inline CMakeLists.txt]: You are of cours totally right. I had used a build script that was using the compiler flags for a debug build 
I'll remove ``-fno-inline`` again for RELEASE and then do another rebase - now that I know how it works 😉

**Comment by oberon2007** [comment]
Done.

**Comment by lasconic** [comment]
Thank you.
